spec/ContentDelegateSpec.js

```diff
@@ -760,6 +766,44 @@ describe('The ContentDelegate class', function() {
     });
   });
 
+  describe('onDownloadAllSubmit', function() {
+    const pre = ('<head><title>test</title><style>body { margin: 0; } iframe { border: none; }' +
+      '</style></head><body>');
+    const iframe = '<iframe src="data:pdf"';
+    const post = ' width="100%" height="100%"></iframe></body>';
+    const html = pre + iframe + post;
+    const cd = zipFileContentDelegate;
+    const eventUrl = "";
+
+    beforeEach(function() {
+      window.chrome = {
+        storage: {
+          local: {
+            get : jasmine.createSpy().and.callFake(function(
+              _, cb) { cb({options : {recap_enabled: true}}); }),
+            set : jasmine.createSpy('set').and.callFake(function() {})
+          }
+        }
+      }
+      cd.onDownloadAllSubmit(eventUrl)
+    });
+
+    it('fetches the page html and extracts the zipFile url', function() {});
+
+    it('downloads the zipFile and stores it in chrome storage', function() {});
+
+    it('checks options to see if recap is enabled', function() {});
+
+    it('uploads the Zip file to RECAP', function() {});
+
+    it('redirects the user to the download page and forwards the zip file', function() {})
+
+    it('calls the notifier once the upload finishes', function() {
+      expect(cd.notifier.showUpload).toHaveBeenCalled();
+    });
```

Comment:
I presume this is still on your todo list, but highlighting anyway so it gets on mine. If you want to do a different testing approach because Jasmine is old or whatever, I'm entirely open to that.